src/domain/products/edit/sections/variants/edit-variant-modal.tsx

```diff
@@ -0,0 +1,153 @@
+import { Product, ProductVariant } from "@medusajs/medusa"
+import React from "react"
+import { useForm } from "react-hook-form"
+import Button from "../../../../../components/fundamentals/button"
+import Modal from "../../../../../components/molecules/modal"
+import { countries } from "../../../../../utils/countries"
+import VariantForm, { VariantFormType } from "../../../components/variant-form"
+import useEditProductActions from "../../hooks/use-edit-product-actions"
+import { createAddPayload } from "./add-variant-modal"
+import { createUpdatePayload } from "./edit-variants-modal/edit-variant-screen"
+
+type Props = {
+  onClose: () => void
+  product: Product
+  variant: ProductVariant
+  isDuplicate?: boolean
+}
+
+const EditVariantModal = ({
+  onClose,
+  product,
+  variant,
+  isDuplicate = false,
+}: Props) => {
+  const form = useForm<VariantFormType>({
+    defaultValues: getEditVariantDefaultValues(variant, product),
+  })
+
+  const {
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+  } = form
+
+  const handleClose = () => {
+    reset(getEditVariantDefaultValues(variant, product))
+    onClose()
+  }
+
+  const {
+    onUpdateVariant,
+    onAddVariant,
+    addingVariant,
+    updatingVariant,
+  } = useEditProductActions(product.id)
+
+  const onSubmit = handleSubmit((data) => {
+    if (isDuplicate) {
```

Comment:
todo: Upon duplicating a product, you cannot save and close before opening the pricing accordion at least once. Ideally, the duplicated product variant will inherit prices as well, so you can do "Duplicate" -> "Save". Wdyt?

Reply:
Fixed, had the save button disabled while the form was not dirty, this is of course not the correct behavior for a duplicate situation. Should be good now!